### PR TITLE
chore: use correct type in `getAccount`

### LINKED ISSUE #9637: Use correct rpc account type for eth_getAccount
### Describe the feature

use correct account type in rpc #9632

blocked by https://github.com/alloy-rs/alloy/pull/1075

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mvares**
Closes #9637

### CODE REVIEW

File: crates/primitives-traits/src/account.rs [hunk 1]
```diff
@@ -8,6 +8,7 @@ use reth_codecs::{reth_codec, Compact};
 use revm_primitives::{AccountInfo, Bytecode as RevmBytecode, JumpTable};
 use serde::{Deserialize, Serialize};
 
+pub use alloy_consensus::Account as AccountTrie;
```
Comment (mvares) [inline crates/primitives-traits/src/account.rs:11]: Sincerely, I don't think this name is idiomatic. I think we can found a better name for it 👍🏻

Comment (mattsse) [inline crates/primitives-traits/src/account.rs:11]: I don't like this export, please remove

File: crates/primitives-traits/src/lib.rs [hunk 2]
```diff
@@ -22,7 +22,7 @@ pub use constants::gas_units::{format_gas, format_gas_throughput};
 
 /// Minimal account
 pub mod account;
-pub use account::{Account, Bytecode};
+pub use account::{Account, AccountTrie, Bytecode};
```
Comment (mattsse) [inline crates/primitives-traits/src/lib.rs:25]: same

File: crates/rpc/rpc-eth-api/src/core.rs [hunk 3]
```diff
@@ -4,8 +4,8 @@
 use alloy_dyn_abi::TypedData;
 use jsonrpsee::{core::RpcResult, proc_macros::rpc};
 use reth_primitives::{
-    transaction::AccessListResult, Account, Address, BlockId, BlockNumberOrTag, Bytes, B256, B64,
-    U256, U64,
+    transaction::AccessListResult, AccountTrie, Address, BlockId, BlockNumberOrTag, Bytes, B256,
```
Comment (mattsse) [inline crates/rpc/rpc-eth-api/src/core.rs:7]: this account type should be available via rpc-types crate

File: crates/rpc/rpc-eth-api/src/core.rs [hunk 4]
```diff
@@ -250,7 +250,7 @@ pub trait EthApi {
 
     /// Returns the account details by specifying an address and a block number/tag
     #[method(name = "getAccount")]
-    async fn get_account(&self, address: Address, block: BlockId) -> RpcResult<Account>;
+    async fn get_account(&self, address: Address, block: BlockId) -> RpcResult<AccountTrie>;
```
Comment (mattsse) [inline crates/rpc/rpc-eth-api/src/core.rs:253]:
```suggestion
    async fn get_account(&self, address: Address, block: BlockId) -> RpcResult<reth_rpc_types::Account>;
```

File: crates/rpc/rpc-eth-api/src/core.rs [hunk 5]
```diff
@@ -626,7 +626,7 @@ where
     }
 
     /// Handler for: `eth_getAccount`
-    async fn get_account(&self, _address: Address, _block: BlockId) -> RpcResult<Account> {
+    async fn get_account(&self, _address: Address, _block: BlockId) -> RpcResult<AccountTrie> {
```
Comment (mattsse) [inline crates/rpc/rpc-eth-api/src/core.rs:629]:
```suggestion
    async fn get_account(&self, _address: Address, _block: BlockId) -> RpcResult<reth_rpc_types::Account> {
```

**Review by mattsse — CHANGES_REQUESTED** [review]

**Review by mattsse — APPROVED** [review]

**Comment by mattsse** [comment]
only needs cargo +nightly fmt

**Pull request merged**